### PR TITLE
Add compliance CI to stackrox/stackrox repo

### PR DESCRIPTION
**Pull request by rhmdnd**
Add a job for running dedicated e2e compliance tests when compliance-related code is changed in stackrox/stackrox.

This change goes hand-in-hand with https://github.com/stackrox/stackrox/pull/6829

### CODE REVIEW

**Comment by rhmdnd** [comment]
Once https://github.com/stackrox/stackrox/pull/6829 merges we can start to rehearse the job here (using `pj-rehearse`)

**Comment by rhmdnd** [comment]
/pj-rehearse pull-ci-stackrox-stackrox-master-ocp-4-10-compliance-e2e-tests

**Comment by rhmdnd** [comment]
/pj-rehearse pull-ci-stackrox-stackrox-master-ocp-4-10-compliance-e2e-tests

**Comment by rhmdnd** [comment]
The rehearsal jobs are failing consistently with:

```
 INFO: Thu Jul 13 21:46:05 UTC 2023: ocp-compliance-e2e-tests will run because this is an openshift/release PR
ERROR: There is no job script for ocp-compliance-e2e-tests 
```

Which is interesting given we landed the script in stackrox with https://github.com/stackrox/stackrox/commit/f0a51597676fdde7e5bd638b215e154af16bb24c

**Comment by rhmdnd** [comment]
> The rehearsal jobs are failing consistently with:
> 
> ```
> INFO: Thu Jul 13 21:46:05 UTC 2023: ocp-compliance-e2e-tests will run because this is an openshift/release PR
> ERROR: There is no job script for ocp-compliance-e2e-tests 
> ```
> 
> Which is interesting given we landed the script in stackrox with [stackrox/stackrox@f0a5159](https://github.com/stackrox/stackrox/commit/f0a51597676fdde7e5bd638b215e154af16bb24c)
> 
> Rebasing this PR to make sure we're really using the latest master branch of stackrox. Otherwise - this job should be getting invoked through https://github.com/stackrox/stackrox/blob/master/.openshift-ci/dispatch.sh#L56-L62.

Actually - the `release-x.y` CI is failing as expected since the script didn't exist there, yet.

@gavin-stackrox just rebased the `release-x.y` branch on master, so that will probably help.

https://github.com/stackrox/stackrox/blob/release-x.y/scripts/ci/jobs/ocp_compliance_e2e_tests.py

**Comment by rhmdnd** [comment]
/pj-rehearse pull-ci-stackrox-stackrox-release-x.y-ocp-4-10-compliance-e2e-tests

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml [hunk 1]
```diff
@@ -366,6 +366,79 @@ postsubmits:
...
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+      timeout: 5h0m0s
+    labels:
+      ci-operator.openshift.io/cloud: aws
+      ci-operator.openshift.io/cloud-cluster-profile: aws
+      ci-operator.openshift.io/variant: ocp-4-10
+      ci.openshift.io/generator: prowgen
+    max_concurrency: 1
```
Comment (gavin-stackrox) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml:383]:
```suggestion
    max_concurrency: 6
```

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml [hunk 2]
```diff
@@ -585,6 +658,83 @@ postsubmits:
...
+    decorate: true
+    decoration_config:
+      skip_cloning: true
+      timeout: 5h0m0s
+    labels:
+      ci-operator.openshift.io/cloud: gcp
+      ci-operator.openshift.io/cloud-cluster-profile: gcp
+      ci-operator.openshift.io/variant: ocp-4-13
+      ci.openshift.io/generator: prowgen
+    max_concurrency: 1
```
Comment (gavin-stackrox) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml:675]:
```suggestion
    max_concurrency: 6
```

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]

**Comment by rhmdnd** [comment]
/pj-rehearse pull-ci-stackrox-stackrox-master-ocp-4-10-compliance-e2e-tests
/pj-rehearse pull-ci-stackrox-stackrox-master-ocp-4-13-compliance-e2e-tests
/pj-rehearse pull-ci-stackrox-stackrox-release-x.y-ocp-4-10-compliance-e2e-tests
/pj-rehearse pull-ci-stackrox-stackrox-release-x.y-ocp-4-13-compliance-e2e-tests

**Comment by rhmdnd** [comment]
/pj-rehearse pull-ci-stackrox-stackrox-release-x.y-ocp-4-13-compliance-e2e-tests
/pj-rehearse pull-ci-stackrox-stackrox-master-ocp-4-13-compliance-e2e-tests

**Comment by rhmdnd** [comment]
/pj-rehearse pull-ci-stackrox-stackrox-master-ocp-4-13-compliance-e2e-tests

**Comment by rhmdnd** [comment]
Running the last rehearsal job and this should be good to go.

**Review by gavin-stackrox — APPROVED** [review]
/lgtm
/honk

**Pull request merged**